src/background/modules/AutocorrectHandler.js

```diff
@@ -26,6 +26,76 @@ let antipatterns = [];
 // Adapted from: https://github.com/mozilla/webextension-polyfill/blob/master/src/browser-polyfill.js
 const IS_CHROME = Object.getPrototypeOf(browser) !== Object.prototype;
 
+/**
+ * Traverse Trie tree of objects to create RegEx.
+ *
+ * @param {Object.<string, Object|boolean>} obj
+ * @returns {string}
+ */
+function pattern(obj) {
+    const alt = [];
+    const cc = [];
+
+    // Escape special characters
+    const regExSpecialChars = /[.*+?^${}()|[\]\\]/gu;
+
+    for (const char in obj) {
+        if (char) {
+            const achar = char.replace(regExSpecialChars, "\\$&");
+
+            if (!("" in obj[char] && Object.keys(obj[char]).length === 1)) {
+                const recurse = pattern(obj[char]);
+                alt.push(recurse + achar);
+            } else {
+                cc.push(achar);
+            }
+        }
+    }
+
+    if (cc.length) {
+        alt.push(cc.length === 1 ? cc[0] : `[${cc.join("")}]`);
+    }
+
+    let result = alt.length === 1 ? alt[0] : `(?:${alt.join("|")})`;
+
+    if ("" in obj) {
```

Comment:
uhm why?

This is really hard to understand, maybe either rexpand the JSDOC description of a high-level overview of the algorithm or somehow refactor it.
(To brainstorm: Maybe we can e.g. have an actual class `TriTree` with operators on it. Somehow `leaf` or `children` should be named or so.)

Reply:
This checks if the current node in the tree is a leaf node, although I am somewhat abusing the "leaf" term. It usually means a node with no children (like a 🍁 in a 🌳), but I am referring a node that marks the end of one or more autocorrections.

In a previous iteration of this function for my Server Status and Link Creator add-ons, I actually used "leaf" for the key (see [here](https://github.com/tdulcet/Server-Status/blob/26c618ec119fef2ffbbf9cfa7285d7e403ba2e42/background.js#L731)), which may have been more readable, but it creates a problem since the string "leaf" could be part of an autocorrection.

An overview of the Trie algorithm can be found on Wikipedia: https://en.wikipedia.org/wiki/Trie. I think a JS Class is a bit excessive for a relatively simple function that is only used once, but I am of course open to ideas of how to make it more understandable.

Reply:
Yeah it is used only once _now_ and the linked JS class was quite simple (though I'd strip unused functions), but it's fine if we get it more readable inside this function.